web_decimal_numpad_dot/static/src/js/numpad_dot.js

```diff
@@ -0,0 +1,75 @@
+/* License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). */
+
+odoo.define('web_decimal_numpad_dot.FieldFloat', function (require) {
+"use strict";
+
+    var core = require('web.core');
+    var form_widgets = require('web.form_widgets');
+    var translation = require('web.translation');
+
+    var FieldFloatDecimalDot = form_widgets.FieldFloat.extend({
+        events: {
+            'keypress': function (e) {
+                if (e.which === 46 || e.which === 44) {
+                    e.preventDefault();
+                    var input = this.$input || this.$('input');
+                    var l10n = translation._t.database.parameters;
+                    if (!_.str.contains(input.val(), l10n.decimal_point)) {
+                        try {
+                            var caret_pos = input[0].selectionStart;
+                            var selection_end = input[0].selectionEnd;
+                            var cur_val = input.val();
+                            var newval = cur_val.substring(0, caret_pos) + l10n.decimal_point + cur_val.substring(selection_end);
+                            input.val(newval);
+                            input[0].selectionStart = input[0].selectionEnd = caret_pos + 1;
+                        } catch (err) {
+                            //fallback to appending if no caret position can be determined
+                            input.val(input.val() + l10n.decimal_point);
+                        }
+                    }
+                }
+            }
+        }
+    });
+    core.form_widget_registry.add('float', FieldFloatDecimalDot);
+    return {
+        FieldFloat: FieldFloatDecimalDot
+    };
+});
+
+odoo.define('web_decimal_numpad_dot.FieldMonetary', function (require) {
+"use strict";
+
+    var core = require('web.core');
+    var form_widgets = require('web.form_widgets');
+    var translation = require('web.translation');
+
+    var FieldMonetaryDecimalDot = form_widgets.FieldMonetary.extend({
+        events: {
+            'keypress': function (e) {
```

Comment:
I can see here that the definition is mostly the same as in the above `FieldFloat` widget.

AFAICS, [`FieldMonetary` extends `FieldFloat`](https://github.com/odoo/odoo/blob/cd1a8f625276e3c40b320df73e4041004c77aea8/addons/web/static/src/js/views/form_widgets.js#L1441), so, if you use the `.include` method in `FieldFloat`, possibly all will get included automatically in `FieldMonetary`.

Please check that out, and if it is the case, just remove all your `FieldMonetary` stuff. If it is not the case, then the common pattern is to do a mixin and mix it twice:

```js
NumpadDotMixin = {
  "events": ...
};

form_widgets.FieldFloat.include(NumpadDotMixin);
form_widgets.FieldMonetary.include(NumpadDotMixin);
```